src/base/bittorrent/trackerentry.h

```diff
@@ -28,17 +28,20 @@
 
 #pragma once
 
-#include <libtorrent/announce_entry.hpp>
-
 #include <QtGlobal>
-
-class QString;
+#include <QString>
+#include <QVector>
 
 namespace BitTorrent
 {
-    class TrackerEntry
+    struct TrackerEntry
     {
-    public:
+        QString url;
+        int tier = 0;
+
+        TrackerEntry() = default;
+        TrackerEntry(const QString &url, int tier = 0);
+
         enum Status
```

Comment:
I would put the `enum` & `struct Endpoint` declarations at the top of the `struct TrackerEntry` body.